src/bin/tree-sitter-graph/main.rs

```diff
@@ -67,6 +76,17 @@ fn main() -> Result<()> {
     let tree = parser
         .parse(&source, None)
         .ok_or_else(|| anyhow!("Could not parse {}", source_path.display()))?;
+    let allow_parse_errors = matches.is_present("allow-parse-errors");
+    if !allow_parse_errors {
+        let parse_errors = ParseErrors::from_tree(&tree, &source);
```

Comment:
I wonder if it would be a bit more smooth to have `from_tree` return an `Option<ParseError>` (that could then be decomposed in an `if let`). That way, you wouldn't need `is_empty` at all. 

Though of course then you have the awkwardness of `Some(vec![])` being a potential value that should never appear.

Hm... Maybe it would suffice to just display the _first_ syntax error in the file? (Which I believe will be the first error node encountered during the tree walk.) After all, after the first syntax error, the rest of the parse could be completely out of whack.